### ISSUE TITLE
Suggestion: Detect invalid number of arguments to 'ln -s'

### ISSUE THREAD

**Issue by nihilus** [koalaman/shellcheck#1080]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/your/interpreter
ln -s "$foo $bar"
```

#### Here's what shellcheck currently says:
Nothing

#### Here's what I wanted or expected to see:
Some sort of error output.

**Comment by Herst** [comment]
Non-POSIX but still quite common: https://unix.stackexchange.com/questions/219038/why-does-ln-s-accept-a-single-argument

**Comment by jleffler** [comment]
This request is no longer checking the shell script for shell scripting issues; it is checking the invocation of command line programs.  That smacks of feature creep, especially given the observation that the specific `ln` notation is quite widely supported.

More particularly, where does this stop.  Will you require shell check to monitory `mv` and `cp` too?  What about `chmod`?  Can you check `ls`?  What about `find`?

I don't think this is something shellcheck should be trying to handle.

**Comment by Herst** [comment]
@jleffler Shellcheck already when into that direction (giving warnings about tools which aren't common shell builtins), see e.g.:
* https://github.com/koalaman/shellcheck/wiki/SC2012
* https://github.com/koalaman/shellcheck/wiki/SC2024
* https://github.com/koalaman/shellcheck/wiki/SC2185
* https://github.com/koalaman/shellcheck/wiki/SC2186

**Comment by koalaman** [comment]
ShellCheck definitely aims to warn about common bad usage patterns of external commands, but it does not intend to generally validate usage.

Whether or not this warrants a check depends on how commonly it occurs, what the false positive rate would be, and how hard it is to solve the problem without ShellCheck's help (`cp foo` gives a clear message, but I had no idea `ln foo` was a valid invocation).

I'll have a look at a corpus of scripts to see how commonly this occurs.

**Comment by koalaman** [comment]
This had an occurrence of less than 1 in 1000 scripts, but there were some interesting and hard-to-spot cases and no false positives among the thousands of cp/ln/mv statements, so I added checks. Thanks!

**Comment by nihilus** [comment]
Worked like a charm over here, good job! And it error'ed on the faulty use as well.

**Comment by nihilus** [comment]
@koalaman seems like you missed to update the ChangeLog fwith this fix.

**Comment by koalaman** [comment]
Thanks, I've updated the changelog in 1d0db92. I'm still getting used to having one.
